### PR TITLE
Zip

### PR DESCRIPTION
**Pull request by muselesscreator**
Updates integration tests and adds zip download functionality.

Requires zipped sample files from [here](https://drive.google.com/file/d/1OEHObEIRA8NywKnz6bJpSMhrJ_SL3bUY/view?usp=sharing) be extracted as `public/samples`

JIRA: 
[AU-457](https://openedx.atlassian.net/browse/AU-457)
[AU-456](https://openedx.atlassian.net/browse/AU-456) (partial)

image

https://user-images.githubusercontent.com/5533134/147290127-f8e169ba-31eb-429b-80c2-d26d55581573.mov

### CODE REVIEW

File: src/data/redux/thunkActions/download.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+export const genManifest = (files) => files.map(
+  (file) => `Filename: ${file.name}\nDescription: ${file.description}`,
+);
+
+/**
+ * Returns the zip filename
+ * @return {string} - zip download file name
+ */
+export const zipFileName = () => {
+  const currentDate = new Date().getTime();
```
Comment (leangseu-edx) [inline src/data/redux/thunkActions/download.js:24]: Should we format this?

File: src/data/redux/thunkActions/download.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+ */
+export const downloadBlobs = (files) => Promise.all(files.map(module.downloadFile));
+
+/**
+ * Download all files for the selected submission as a zip file.
+ * Throw error and do not download zip if any of the files fail to fetch.
+ */
+export const downloadFiles = () => (dispatch, getState) => {
+  const { files } = selectors.grading.selected.response(getState());
+  dispatch(networkRequest({
```
Comment (leangseu-edx) [inline src/data/redux/thunkActions/download.js:65]: If one of the files fail to download, we are going to fail the entire things? Should we update the manifest instead? We do not have alternative for them to get the files if one of the files somehow fail.

File: src/data/redux/thunkActions/download.js (hunk 1 above)
Comment (muselesscreator) [reply src/data/redux/thunkActions/download.js:24]: 100%  Just waiting for a real name from product.  is placeholder, hence why no test

File: src/data/redux/thunkActions/download.js (hunk 2 above)
Comment (muselesscreator) [reply src/data/redux/thunkActions/download.js:65]: Eh... I would prompt product for that question.  That bloats this feature not-insignificantly.  Could be a future-feature ticket tho?

Comment (jnlapierre) [reply src/data/redux/thunkActions/download.js:65]: It has been decided that we are going to fail the entire thing for MVP.

**Review by jnlapierre — APPROVED** [review]
Installed locally, manually tested and confirmed, all automated tests pass LGTM

**Pull request merged**